src/var/isFunction.js

```diff
@@ -7,7 +7,8 @@ define( function() {
 		// In some browsers, typeof returns "function" for HTML <object> elements
 		// (i.e., `typeof document.createElement( "object" ) === "function"`).
 		// We don't want to classify *any* DOM node as a function.
-		return typeof obj === "function" && typeof obj.nodeType !== "number";
+		return typeof obj === "function" && typeof obj.nodeType !== "number" &&
+			[ "[object HTMLCollection]", "[object NodeList]" ].indexOf( obj.toString() ) === -1;
```

Comment:
Are you certain these are the only two needed additions?

Reply:
I believe these are the only two really needed.
- `NodeList` is returned when we query elements on `document`.
- `HTMLCollection` is returned when we access node `children` property.


**My investigation**
On Mozilla website (https://developer.mozilla.org/en-US/docs/Web/API) under _**Interfaces**_ section I can see many different `*List` or `*Collection` types. However such collections have to be accessed in a special non generic way.
For example `<select>` tag could return different result depending on how we access its children nodes:
- Generic way to access children
```javascript
> $("select")[0].children.toString()
"[object HTMLCollection]"
```
- Specific way, and it only works because element type is `HTMLSelectElement`
```javascript
> $("select")[0].options.toString()
"[object HTMLOptionsCollection]"
```
@mgol please let me know if you are aware of jQuery accessing DOM elements other way than just querying `document` or accessing `children` property? If so then we should also cover such cases. Otherwise I would keep just these two.

Reply:
Not applicable anymore due to new, more generic solution i.e. `obj.item !== "function`.